### PR TITLE
More MPI serialization support

### PR DESCRIPTION
**Pull request by akva2**
Downstream of https://github.com/OPM/opm-common/pull/1353

### CODE REVIEW

**Comment by bska** [comment]
Other than the fact that this looks to be following the pattern established earlier I can't really comment.  I would prefer if the test assertions of the form
```C++
BOOST_CHECK(condition);
```
instead were phrased as
```C++
BOOST_CHECK_MESSAGE(condition, "Contextual failure message");
```
but I suppose we could update those in a future PR.

File: tests/test_ParallelRestart.cpp [hunk 1]
```diff
@@ -571,6 +571,17 @@ BOOST_AUTO_TEST_CASE(Tabdims)
 }
 
 
+BOOST_AUTO_TEST_CASE(EndpointScaling)
+{
+#if HAVE_MPI
+    Opm::EndpointScaling val1(std::bitset<4>(123456));
```
Comment (bska) [inline tests/test_ParallelRestart.cpp]: This is an unexpected initial value for a bitset of size 4.  From reference material I know that it's perfectly well defined (and creates an all-ones bitset), but I still had to think twice about it.

Comment (akva2) [reply tests/test_ParallelRestart.cpp]: right, i can make it a valid value.

**Comment by akva2** [comment]
i really would prefer to do those afterwards, every change I have to make give me a lot of conflicts to resolve in the remaining patch set of ~ 130 patches..

File: tests/test_ParallelRestart.cpp (hunk 1 above)
Comment (bska) [reply tests/test_ParallelRestart.cpp]: > right, i can make it a valid value.

Cool.  You might even consider constructing the bitset from a string:
```C++
std::bitset<4>("1111")
```

**Comment by bska** [comment]
> i really would prefer to do those afterwards

Very well.  Let's do that.

> every change I have to make give me a lot of conflicts to resolve in the remaining patch set of ~ 130 patches..

Right.  Let's get those in first and tidy up in a separate pass later.

**Pull request merged**